ibr/binToJs.js

```diff
@@ -111,52 +16,9 @@ function onChooseFile() {
   if (file) {
     const fr = new FileReader();
     fr.onload = function(evt) {
-      const ibrData = InternalBuildingRepresentation.read(
-          new Pbf( evt.target.result ));
-      const scene = new THREE.Scene();
-      const camera = new THREE.PerspectiveCamera(
-          75, window.innerWidth / window.innerHeight, 0.1, 10000 );
-      const renderer = new THREE.WebGLRenderer();
-      renderer.setSize( window.innerWidth, window.innerHeight );
-      document.body.appendChild( renderer.domElement );
-      const controls = new OrbitControls( camera, renderer.domElement );
-      controls.target.set( 0, 0.5, 0 );
-      controls.update();
-      controls.enablePan = false;
-      controls.enableDamping = true;
-
-      // for datafiles that have top level name is ""
-      if (ibrData.name === '') {
-        ibrData.name = 'ibrData.name';
-      }
-      const li = document.createElement('li'); // list element container
-      document.getElementById('layerList').appendChild(li);
-      const rootSpan = createLabel('span', ibrData.name, li);
-      rootSpan.setAttribute('class', 'arrow');
-      li.appendChild(rootSpan);
-      const ul = document.createElement('ul');
-      ul.setAttribute('class', 'nested');
-      ul.setAttribute('id', ibrData.name);
-      li.appendChild(ul);
-      rootSpan.addEventListener('click', function() {
-        rootSpan.parentElement.querySelector('.nested').classList
-            .toggle('active');
-        rootSpan.classList.toggle('expanded-arrow');
-      });
-      extractSingleStructureData(ibrData, ibrData.name, scene);
-
-      camera.position.set( 0, 0, 7000 );
-      camera.lookAt( 0, 0, 0 );
-      animate();
-
-      /**
-       * Renders the scene.
-       */
-      function animate() {
-        requestAnimationFrame( animate );
-        controls.update();
-        renderer.render( scene, camera );
-      }
+      const scene = IBRSDK.generateScene(document.body);
```

Comment:
can we be more specific than document.body? users might want to have more control over where they hook the scene into.

maybe pass in a document.getElementById('ibr-scene')?

Reply:
used element 'mainCanvas' as parent element for all THREE.js rendering